test/unsafekzg/kzgsrs.go

```diff
@@ -0,0 +1,393 @@
+// Package unsafekzg is a convenience package (to be use for test purposes only)
+// to generate and cache SRS for the kzg scheme (and indirectly for PlonK setup).
+//
+// Functions in this package are thread safe.
+package unsafekzg
+
+import (
+	"bufio"
+	"crypto/rand"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark-crypto/kzg"
+	"github.com/consensys/gnark/constraint"
+	"github.com/consensys/gnark/internal/utils"
+
+	kzg_bls12377 "github.com/consensys/gnark-crypto/ecc/bls12-377/kzg"
+	kzg_bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381/kzg"
+	kzg_bls24315 "github.com/consensys/gnark-crypto/ecc/bls24-315/kzg"
+	kzg_bls24317 "github.com/consensys/gnark-crypto/ecc/bls24-317/kzg"
+	kzg_bn254 "github.com/consensys/gnark-crypto/ecc/bn254/kzg"
+	kzg_bw6633 "github.com/consensys/gnark-crypto/ecc/bw6-633/kzg"
+	kzg_bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761/kzg"
+
+	fft_bls12377 "github.com/consensys/gnark-crypto/ecc/bls12-377/fr/fft"
+	fft_bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381/fr/fft"
+	fft_bls24315 "github.com/consensys/gnark-crypto/ecc/bls24-315/fr/fft"
+	fft_bls24317 "github.com/consensys/gnark-crypto/ecc/bls24-317/fr/fft"
+	fft_bn254 "github.com/consensys/gnark-crypto/ecc/bn254/fr/fft"
+	fft_bw6633 "github.com/consensys/gnark-crypto/ecc/bw6-633/fr/fft"
+	fft_bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761/fr/fft"
+
+	fr_bls12377 "github.com/consensys/gnark-crypto/ecc/bls12-377/fr"
+	fr_bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381/fr"
+	fr_bls24315 "github.com/consensys/gnark-crypto/ecc/bls24-315/fr"
+	fr_bls24317 "github.com/consensys/gnark-crypto/ecc/bls24-317/fr"
+	fr_bn254 "github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	fr_bw6633 "github.com/consensys/gnark-crypto/ecc/bw6-633/fr"
+	fr_bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761/fr"
+
+	bls12377 "github.com/consensys/gnark-crypto/ecc/bls12-377"
+	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	bls24315 "github.com/consensys/gnark-crypto/ecc/bls24-315"
+	bls24317 "github.com/consensys/gnark-crypto/ecc/bls24-317"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	bw6633 "github.com/consensys/gnark-crypto/ecc/bw6-633"
+	bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761"
+)
+
+var (
+	cache           map[uint64]cacheEntry
+	memLock, fsLock sync.RWMutex
+)
+
+func init() {
+	cache = make(map[uint64]cacheEntry)
```

Comment:
Do not necessarily need init() function. Can just define global var `cache = make(map[uint64]cacheEntry)`. For example see https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/database/sql/sql.go;l=35